syncserver-common/src/lib.rs

```diff
@@ -121,16 +121,15 @@ impl BlockingThreadpool {
         E: fmt::Debug + Send + InternalError + 'static,
     {
         self.spawned_tasks.fetch_add(1, Ordering::Relaxed);
+        scopeguard::defer! {
```

Comment:
Please add a comment here noting why we're waiting until we run the task after the function goes out of scope.